src/com/dotcms/api/system/event/Payload.java

```diff
@@ -80,9 +91,17 @@ public Visibility getVisibility() {
 
 	/**
 	 * Returns the visibility id
-	 * @return String
+	 * @return Object
      */
 	public String getVisibilityId() {
 		return visibilityId;
 	}
+
+	public boolean verified(SessionWrapper session) {
```

Comment:
yeah but the visibilityId is just a data, not functionality.
In the other hand, the PayloadVerifier is a functionality but it is totally up to you, if the serialization works np.